connectors/message/src/main/java/io/camunda/connector/message/SendMessageConnectorFunction.java

```diff
@@ -0,0 +1,131 @@
+
+  private static final Logger LOG = LoggerFactory.getLogger(SendMessageConnectorFunction.class);
+
+  @Override
+  public Object execute(OutboundConnectorContext context) throws Exception {
+
+    SendMessageRequest messageRequest = context.bindVariables(SendMessageRequest.class);
+    LOG.debug(
+        "Invoke send message connector with name {} and correlation key {}",
+        messageRequest.messageName(),
+        messageRequest.correlationKey());
+
+    ZeebeClient zeebeClient = ZeebeClientContext.getZeebeClient();
+
+    if ("publish".equals(messageRequest.mode())) {
+      PublishMessageCommandStep3 publishMessageCommand;
+      PublishMessageCommandStep2 step2 =
+          zeebeClient.newPublishMessageCommand().messageName(messageRequest.messageName());
+      if (messageRequest.correlationKey() != null
+          && messageRequest.correlationKey().isBlank() == false) {
+        publishMessageCommand = step2.correlationKey(messageRequest.correlationKey());
+      } else {
+        publishMessageCommand = step2.withoutCorrelationKey();
+      }
+      if (messageRequest.messageId() != null) {
+        publishMessageCommand.messageId(messageRequest.messageId());
+      }
+      if (messageRequest.variables() != null) {
+        publishMessageCommand.variables(messageRequest.variables());
+      }
+      if (messageRequest.timeToLive() != null) {
+        publishMessageCommand.timeToLive(messageRequest.timeToLive());
+      }
+      if (messageRequest.tenantId() != null) {
+        publishMessageCommand.tenantId(messageRequest.tenantId());
+      }
+      if (messageRequest.requestTimeout() != null) {
+        publishMessageCommand.requestTimeout(messageRequest.requestTimeout());
+      }
+      PublishMessageResponse publishMessageResponse = publishMessageCommand.send().join();
+      LOG.debug("message published with messageKey {}", publishMessageResponse.getMessageKey());
+      return publishMessageResponse;
+    } else if ("correlate".equals(messageRequest.mode())) {
+      CorrelateMessageCommandStep2 correlateMessageCommand =
+          zeebeClient.newCorrelateMessageCommand().messageName(messageRequest.messageName());
+      CorrelateMessageCommandStep3 correlateMessageCommandStep3;
+      if (messageRequest.correlationKey() == null || messageRequest.correlationKey().isBlank()) {
+        correlateMessageCommandStep3 = correlateMessageCommand.withoutCorrelationKey();
+      } else {
+        correlateMessageCommandStep3 =
+            correlateMessageCommand.correlationKey(messageRequest.correlationKey());
+      }
+      if (messageRequest.variables() != null) {
+        correlateMessageCommandStep3.variables(messageRequest.variables());
+      }
+      if (messageRequest.tenantId() != null) {
+        correlateMessageCommandStep3.tenantId(messageRequest.tenantId());
+      }
+      if (messageRequest.requestTimeout() != null) {
+        correlateMessageCommandStep3.requestTimeout(messageRequest.requestTimeout());
+      }
```

Comment:
What about having a separate Mapper class the is repsonsible for mapping a `SendMessageRequest` to a `Command`? 

This woudl make this code more readable and keep the Connector function simple.

Reply:
Done. Looks better.